From 76f0ede2e921e63518205564b903408aea3350c6 Mon Sep 17 00:00:00 2001
From: Matt Dembski <dembski@google.com>
Date: Tue, 21 May 2024 20:14:26 +0000
Subject: [PATCH] Fix crash in CookieControlsController::OnEntryPointAnimated()

The crash happens when the WebContents is not prepared before the animation is called. The fix is to check if the WebContents is valid before accessing it.

Bug:341576798

Change-Id: I3ae02be88d915b6ec1e3381419a749ee70c09ec1
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5550353
Reviewed-by: Elly FJ <ellyjones@chromium.org>
Reviewed-by: Fiona Macintosh <fmacintosh@google.com>
Auto-Submit: Matt Dembski <dembski@google.com>
Commit-Queue: Matt Dembski <dembski@google.com>
Cr-Commit-Position: refs/heads/main@{#1304004}
---

--- a/components/content_settings/browser/ui/cookie_controls_controller.cc
+++ b/components/content_settings/browser/ui/cookie_controls_controller.cc
@@ -265,6 +265,12 @@
 }
 
 void CookieControlsController::OnEntryPointAnimated() {
+  // sanity check if WebContents was instantiated (update method called before)
+  // TODO(b/341972754): refactor this to be handled properly via update method
+  // for all Android corner cases.
+  if (GetWebContents() == nullptr) {
+    return;
+  }
   const GURL& url = GetWebContents()->GetLastCommittedURL();
   base::Value::Dict metadata = GetMetadata(settings_map_, url);
   metadata.Set(kEntryPointAnimatedKey, base::Value(true));
